File: boleanos/boleanos.py
print("Tabla de verdad para la compuerta AND:")
print("A   | B   | A AND B")
print("----|-----|-------")
for a in [False, True]:
    for b in [False, True]:
        print(f"{a}   | {b}   | {a and b}")

print("\nTabla de verdad para la compuerta OR:")
print("A   | B   | A OR B")
print("----|-----|------")
for a in [False, True]:
    for b in [False, True]:
        print(f"{a}   | {b}   | {a or b}")

print("\nTabla de verdad para la negación de la compuerta AND:")
print("A   | B   | NOT (A AND B)")
print("----|-----|-------------")
for a in [False, True]:
    for b in [False, True]:
        print(f"{a}   | {b}   | {not (a and b)}")

print("\nTabla de verdad para la negación de la compuerta OR:")
print("A   | B   | NOT (A OR B)")
print("----|-----|------------")
for a in [False, True]:
    for b in [False, True]:
        print(f"{a}   | {b}   | {not (a or b)}")

File: boleanos/boleanos.cpp
#include <iostream>

int main() {
    bool a, b;
    std::cout << "Tabla de verdad para la compuerta AND:" << std::endl;
    std::cout << "A   | B   | A AND B" << std::endl;
    std::cout << "----|-----|-------" << std::endl;
    for (a = false; a <= true; a = !a) {
        for (b = false; b <= true; b = !b) {
            std::cout << a << "   | " << b << "   | " << (a && b) << std::endl;
        }
    }

    std::cout << "\nTabla de verdad para la compuerta OR:" << std::endl;
    std::cout << "A   | B   | A OR B" << std::endl;
    std::cout << "----|-----|------" << std::endl;
    for (a = false; a <= true; a = !a) {
        for (b = false; b <= true; b = !b) {
            std::cout << a << "   | " << b << "   | " << (a || b) << std::endl;
        }
    }

    std::cout << "\nTabla de verdad para la negación de la compuerta AND:" << std::endl;
    std::cout << "A   | B   | NOT (A AND B)" << std::endl;
    std::cout << "----|-----|-------------" << std::endl;
    for (a = false; a <= true; a = !a) {
        for (b = false; b <= true; b = !b) {
            std::cout << a << "   | " << b << "   | " << !(a && b) << std::endl;
        }
    }

    std::cout << "\nTabla de verdad para la negación de la compuerta OR:" << std::endl;
    std::cout << "A   | B   | NOT (A OR B)" << std::endl;
    std::cout << "----|-----|------------" << std::endl;
    for (a = false; a <= true; a = !a) {
        for (b = false; b <= true; b = !b) {
            std::cout << a << "   | " << b << "   | " << !(a || b) << std::endl;
        }
    }

    return 0;
}
